ddprof-lib/src/main/cpp/profiler.cpp

```diff
@@ -932,8 +932,9 @@ void Profiler::updateNativeThreadNames() {
     ThreadList* thread_list = OS::listThreads();
     for (int tid; (tid = thread_list->next()) != -1; ) {
         _thread_info.updateThreadName(tid, [](int tid) -> std::unique_ptr<char[]> {
-            char *name_buf = new char[64];
-            if (OS::threadName(tid, name_buf, sizeof(name_buf))) {
+            const size_t buffer_size = 64;
+            char *name_buf = new char[buffer_size];
```

Comment:
So this is fixed size. The sizeof would have worked if we had created the buffer without a dynamic allocation.
The better practice would have been to use a standard container: std::array and size operators

Reply:
Apologies for being unclear, the suggestions are as follows:
- use a static C style buffer: this is consistent with the coding style of the project, then sizeof would work (or continue using a const or constexpr buffer_size to avoid possible issues)
- use a static C++ style container (std::array): this is not consistent with the coding style (and possibly C++17). This allows you to just start using size operators which would be constexpr.
Feel free to chose either options.